Plutarch/Api/V1.hs

```diff
@@ -431,19 +470,32 @@ instance PIsDataRepr PCredential where
           DRHNil
 
 data PStakingCredential (s :: S)
-  = PStakingHash (Term s (PDataList '[PCredential]))
-  | PStakingPtr (Term s (PDataList '[PInteger, PInteger, PInteger]))
+  = PStakingHash (Term s (PDataRecord '["_0" ':= PCredential]))
+  | PStakingPtr
+      ( Term
+          s
+          ( PDataRecord
+              '[ "_0" ':= PInteger
+               , "_1" ':= PInteger
+               , "_2" ':= PInteger
+               ]
```

Comment:
There is an opportunity to generalize this using a `PDataList` that is computed from `PDataRecord`. You can even redefine `PTuple` in terms of it. Something like:

```haskell
type PDataList (as :: [PType]) s =
  PDataRecord (Indexed as 'Z)

type family Indexed (as :: [PType]) (start :: Nat) :: [PLabeled] where
  Indexed '[] _ = '[]
  -- TODO: convert `Nat` to `Symbol`
  Indexed (a ': as) n = "_?" ':= a ': Indexed as ('S n)

type PTuple' a b s =
  PDataList '[a, b] 'Z s
```

Reply:
I don't think this would be worth it.

Reply:
Now that I think about it, we do _want_ to be explicit here in regards to indices, so as to not inadvertently break decoding of Data. So ignore my suggestion.

This explicitness is enabled in products only, however. Not sums, unfortunately.

Reply:
I tried out something like this, it was more of a headache than is worth it